plugins/inputs/mqtt_consumer/README.md

```diff
@@ -73,12 +73,71 @@ and creates metrics using one of the supported [input data formats][].
   ## more about them here:
   ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_INPUT.md
   data_format = "influx"
+
+  ## Enable extracting tag values from MQTT topics
+  ## _ denotes an ignored entry in the topic path 
+  # [[inputs.mqtt_consumer.topic_parsing]]
+  #   topic = ""
+  #   measurement = ""
+  #   tags = ""
+  #   fields = ""
+  ## Value supported is int, float, unit
+  #   [[inputs.mqtt_consumer.topic.types]]
+  #      key = type
+```
+
+## About Topic Pasring
+
+The MQTT topic as a whole is stored as a tag, but this can be far too coarse
+too be easily used when utilising the data further down the line. This
+change allows tag values to be extracted from the MQTT topic letting you
+store the information provided in the topic in a meaningful way. An _ denotes an
+ignored entry in the topic path. Please see the following example.
+
+## Example Configuration for topic parsing
+
+```toml
+[[inputs.mqtt_consumer]]
+  ## Broker URLs for the MQTT server or cluster.  To connect to multiple
+  ## clusters or standalone servers, use a separate plugin instance.
+  ##   example: servers = ["tcp://localhost:1883"]
+  ##            servers = ["ssl://localhost:1883"]
+  ##            servers = ["ws://localhost:1883"]
+  servers = ["tcp://127.0.0.1:1883"]
+
+  ## Topics that will be subscribed to.
+  topics = [
+    "telegraf/+/cpu/23",
+  ]
+
+  ## Data format to consume.
+  ## Each data format has its own unique set of configuration options, read
+  ## more about them here:
+  ## https://github.com/influxdata/telegraf/blob/master/docs/DATA_FORMATS_INPUT.md
+  data_format = "value"
+  data_type = "float"
+
+  [[inputs.mqtt_consumer.topic_parsing]]
+    topic = "telegraf/one/cpu/23"
+    measurement = "_/_/measurement/_"
+    tags = "tag/_/_/_"
+    fields = "_/_/_/test"
+    [inputs.mqtt_consumer.topic_parsing.types]
+      test = "int"
```

Comment:
Hmmm how about either specifying the type inline separating e.g. using `:`, i.e.
```toml
    fields = "_/_/_/test:int"
```
or having multiple field entries, one per type, i.e.
```toml
    fields_int = "_/_/_/test"
    fields_float = "..."
    ...
```

This would avoid the nesting of toml sections which is probably hard for the user.

Reply:
Thank you for the suggestion, Sebastian and I considered that when we wrote this. 

The nesting of the toml matches the json_v2 parser which is used a lot with mqtt so we were trying to stay consistent. Is there a reason you think the nested table is harder? 

I think that if the user has multiple fields writing something along the lines of `fields = "_/_/testing:float/test:int"` can get complicated.

Reply:
I don't know, but I've seen this here with modbus that people don't always get `[[..]]` and `[...]` right. But it's probably a matter of taste/experience. The other point is the duplication of information, i.e. if you want to rename a field you need to do it in two different places which usually asks for trouble. Anyway, I can also live with your design, just wanted to make the suggestion. :-)

Reply:
Feel free to mark this resolved.